### PR TITLE
Remove Kestrel.Core dependency from Transport.Libuv (#1584)

### PR DESCRIPTION
**Pull request by cesarbs**
#1584

Adds `ILibuvTrace` and `LibuvTrace` so the libuv transport can have its own trace and not depend on `IKestrelTrace`.

### CODE REVIEW

**Comment by halter73** [comment]
Other than removing the comments from IKestrelTrace.cs and removing the unused trace methods from KestrelTrace.cs, what else needs to be done?

**Comment by cesarbs** [comment]
Updated as discussed offline.

For now I've kept the same method names and event IDs.

File: test/Microsoft.AspNetCore.Server.KestrelTests/MultipleLoopTests.cs [hunk 1]
```diff
@@ -7,24 +7,19 @@
...
     {
-        private readonly LibuvFunctions _uv;
-        private readonly IKestrelTrace _logger;
-        public MultipleLoopTests()
-        {
-            _uv = new LibuvFunctions();
-            _logger = new TestKestrelTrace();
-        }
+        private readonly LibuvFunctions _uv = new LibuvFunctions();
+        private readonly ILibuvTrace _logger = Mock.Of<ILibuvTrace>();
```
Comment (halter73) [inline test/Microsoft.AspNetCore.Server.KestrelTests/MultipleLoopTests.cs]: The mock `ILibuvTrace` instances should log to the TestApplicationErrorLogger. Then we can do the `#if true` to `#if false` trick, look at the Messages property, etc...

File: src/Microsoft.AspNetCore.Server.Kestrel.Transport.Libuv/Microsoft.AspNetCore.Server.Kestrel.Transport.Libuv.csproj [hunk 2]
```diff
@@ -14,11 +14,19 @@
...
+    <PackageReference Include="Microsoft.Extensions.Options" Version="$(AspNetCoreVersion)" />
     <PackageReference Include="Microsoft.Extensions.TaskCache.Sources" Version="$(AspNetCoreVersion)" PrivateAssets="All" />
   </ItemGroup>
 
   <ItemGroup>
-    <ProjectReference Include="..\Microsoft.AspNetCore.Server.Kestrel.Core\Microsoft.AspNetCore.Server.Kestrel.Core.csproj" />
+    <ProjectReference Include="..\Microsoft.AspNetCore.Server.Kestrel.Transport.Abstractions\Microsoft.AspNetCore.Server.Kestrel.Transport.Abstractions.csproj" />
+  </ItemGroup>
+
+  <ItemGroup Condition=" '$(TargetFramework)' == 'netstandard1.3' ">
```
Comment (davidfowl) [inline src/Microsoft.AspNetCore.Server.Kestrel.Transport.Libuv/Microsoft.AspNetCore.Server.Kestrel.Transport.Libuv.csproj]: remove net46

**Comment by cesarbs** [comment]
Updated.

File: test/Microsoft.AspNetCore.Server.KestrelTests/UvTimerHandleTests.cs [hunk 3]
```diff
@@ -2,22 +2,23 @@
...
 using Microsoft.AspNetCore.Server.Kestrel.Internal.Networking;
+using Microsoft.AspNetCore.Server.Kestrel.Transport.Libuv.Internal.Infrastructure;
 using Microsoft.AspNetCore.Testing;
 using Xunit;
 
 namespace Microsoft.AspNetCore.Server.KestrelTests
 {
     public class UvTimerHandleTests
     {
+        private readonly ILibuvTrace _trace = new LibuvTrace(new TestApplicationErrorLogger());
```
Comment (davidfowl) [inline test/Microsoft.AspNetCore.Server.KestrelTests/UvTimerHandleTests.cs]: Any specific reason this is class level? Are you embracing xunit?

Comment (cesarbs) [inline test/Microsoft.AspNetCore.Server.KestrelTests/UvTimerHandleTests.cs]: Being consistent with other test classes. I don't like it 😝 But we don't really check any logs in these tests so whatever.

**Pull request merged**